test/functional/inspec_exec_test.rb

```diff
@@ -115,6 +115,26 @@
     out.stdout.must_include "\nTest Summary: \e[38;5;41m2 successful\e[0m, 0 failures, 0 skipped\n"
   end
 
+  it 'does not vendor profiles when using the a local path dependecy' do
+    Dir.mktmpdir do |tmpdir|
+      # Other commands (e.g. vendor) might create a vendor directory on this
+      # shared profile during functional testing. We remove `vendor/` here if
+      # it exists to avoid surprises when checking cache contents later.
+      vendor_dir = File.join(inheritance_profile, 'vendor')
+      FileUtils.remove_dir(vendor_dir) if File.exist?(vendor_dir)
+
+      command = 'exec ' + inheritance_profile + ' --no-create-lockfile'
+      out = inspec_with_env(command, INSPEC_CONFIG_DIR: tmpdir)
+      out.stderr.must_equal ''
+      out.exit_status.must_equal 100
+      out.stdout.must_include "Profile Summary: \e[38;5;41m1 successful control\e[0m, 0 control failures, \e[38;5;247m1 control skipped\e[0m\n"
```

Comment:
If you need to analyze the output of the run, I prefer to use a JSON reporter and parse the output; then you have a clearer test, checking for specific data elements (rather than strings full of ASCII control sequences!)

Just a preference, not a blocker.

Reply:
Hmm...I like that idea. I went with the precedent in that file, but I think your approach makes sense.